### PR TITLE
8324765: C2 SuperWord: remove dead code: SuperWord::insert_extracts

### PR DESCRIPTION
**Pull request by eme64**
Subtask of https://github.com/openjdk/jdk/pull/16620

As far as all tests and code study have showed me, SuperWord::insert_extracts is dead.

I am replacing it with verification code, that checks that no ExtractNode is required.

**Details**

All the relevant cases are marked as "unprofitable" in `SuperWord::profitable`, see:
https://github.com/openjdk/jdk/blob/dfdd2174d7af5e3e995147484db17b45b006f6d0/src/hotspot/share/opto/superword.cpp#L1912-L1915

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8324765](https://bugs.openjdk.org/browse/JDK-8324765): C2 SuperWord: remove dead code: SuperWord::insert_extracts (**Sub-task** - P4)

### Reviewers
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**) ⚠️ Review applies to [8a08a4f2](https://git.openjdk.org/jdk/pull/17589/files/8a08a4f248c95842f4c2651bb542c3aebb0e2453)
 * [Christian Hagedorn](https://openjdk.org/census#chagedorn) (@chhagedorn - **Reviewer**) ⚠️ Review applies to [8a08a4f2](https://git.openjdk.org/jdk/pull/17589/files/8a08a4f248c95842f4c2651bb542c3aebb0e2453)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17589/head:pull/17589` \
`$ git checkout pull/17589`

Update a local copy of the PR: \
`$ git checkout pull/17589` \
`$ git pull https://git.openjdk.org/jdk.git pull/17589/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17589`

View PR using the GUI difftool: \
`$ git pr show -t 17589`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17589.diff">https://git.openjdk.org/jdk/pull/17589.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17589#issuecomment-1912333520)

### CODE REVIEW

**Review by vnkozlov — APPROVED** [review]
Good.

File: src/hotspot/share/opto/superword.cpp [hunk 1]
```diff
@@ -2853,55 +2849,35 @@ Node* SuperWord::vector_opd(Node_List* p, int opd_idx) {
...
 
-//------------------------------insert_extracts---------------------------
-// If a use of pack p is not a vector use, then replace the
-// use with an extract operation.
-void SuperWord::insert_extracts(Node_List* p) {
-  if (p->at(0)->is_Store()) return;
-  assert(_n_idx_list.is_empty(), "empty (node,index) list");
+#ifdef ASSERT
+// We check that every packset (name it p_def) only has vector uses (p_use),
+// which are properly vector uses of def.
```
Comment (chhagedorn) [inline src/hotspot/share/opto/superword.cpp:2854]:
```suggestion
// which are proper vector uses of def.
```

**Review by chhagedorn — APPROVED** [review]
Looks good!

**Comment by eme64** [comment]
Thanks for the reviews @vnkozlov @chhagedorn !
/integrate

**Pull request closed without merging**